src/everything.rs

```diff
@@ -305,8 +307,106 @@ impl Everything {
         self.fileset.fullpath(entry)
     }
 
-    pub fn load_errorkey_config(&self) {
+    pub fn load_config_filtering_rules(&self) {
+        // Load legacy rules:
+        self.load_legacy_ignore_blocks();
+        if self.config.count_keys("filter") > 1 {
+            error(&self.config, ErrorKey::Config, "Detected more than one filter block: there can be only one!");
+        }
+        if let Some(filter) = self.config.get_field_block("filter") {
+            set_show_vanilla(filter.get_field_bool("show_vanilla").unwrap_or(false));
+            set_show_loaded_mods(filter.get_field_bool("show_loaded_mods").unwrap_or(false));
+            let severity = filter
+                .get_field_value("minimum_severity")
+                .map(|t| t.as_str().parse().ok())
+                .flatten()
+                .unwrap_or(Severity::Warning);
+            let confidence = filter
+                .get_field_value("minimum_confidence")
+                .map(|t| t.as_str().parse().ok())
+                .flatten()
+                .unwrap_or(Confidence::Reasonable);
+            set_minimum_level(LogLevel::new(severity, confidence));
+            if let Some(whitelist) = filter.get_field_block("whitelist") {
+                self.load_filter_rules_list(Whitelist, whitelist);
+            }
+            if let Some(blacklist) = filter.get_field_block("blacklist") {
+                self.load_filter_rules_list(Blacklist, blacklist);
+            }
+        }
+    }
+
+    fn load_filter_rules_list(&self, rules_type: RulesListType, rules: &Block) {
+        for (key_opt, comp, value) in rules.iter_items() {
+            if let None = key_opt {
+                error(value, ErrorKey::Config, "Missing key. Loose values are not allowed here.");
+                continue;
+            }
+            let key = key_opt.as_ref().expect("Should exist.");
+            if comp != &Eq {
+                error(key, ErrorKey::Config, &format!("Detected illegal operator `{comp}`, only `=` is allowed."));
+                continue;
+            }
+            match (key.as_str(), &value) {
+                ("key", _) => error(key, ErrorKey::Config, "Detected illegal key: `key` should probably be `rule_key`."),
```

Comment:
Can the keys be defined without the `rule_` prefix in the first place? To keep the config file simpler